meson.build

```diff
@@ -0,0 +1,168 @@
+    'src' / 'Utils' / 'PasswordManager.vala',
+    'src' / 'Utils' / 'Player.vala',
+    'src' / 'Utils' / 'SoupClient.vala',
+    'src' / 'Utils' / 'Utils.vala',
+    'src' / 'Utils' / 'XmlUtils.vala',
+    'src' / 'Vocal.vala',
+    'src' / 'VocalSettings.vala',
+    'src' / 'Widgets' / 'AddFeedDialog.vala',
+    'src' / 'Widgets' / 'ArtworkPopover.vala',
+    'src' / 'Widgets' / 'CoverArt.vala',
+    'src' / 'Widgets' / 'DirectoryArt.vala',
+    'src' / 'Widgets' / 'DirectoryView.vala',
+    'src' / 'Widgets' / 'DownloadDetailBox.vala',
+    'src' / 'Widgets' / 'DownloadsPopover.vala',
+    'src' / 'Widgets' / 'EpisodeDetailBox.vala',
+    'src' / 'Widgets' / 'InternetArchiveUploadDialog.vala',
+    'src' / 'Widgets' / 'NewEpisodesView.vala',
+    'src' / 'Widgets' / 'PlaybackBox.vala',
+    'src' / 'Widgets' / 'PodcastView.vala',
+    'src' / 'Widgets' / 'QueueBox.vala',
+    'src' / 'Widgets' / 'QueueRow.vala',
+    'src' / 'Widgets' / 'SearchResultBox.vala',
+    'src' / 'Widgets' / 'SearchResultsView.vala',
+    'src' / 'Widgets' / 'SettingsDialog.vala',
+    'src' / 'Widgets' / 'Shownotes.vala',
+    'src' / 'Widgets' / 'SyncDialog.vala',
+    'src' / 'Widgets' / 'Toolbar.vala',
+    'src' / 'Widgets' / 'VideoControls.vala',
+]
+
+dependencies = [
+    dependency('clutter-gst-3.0'),
+    dependency('clutter-gtk-1.0'),
+    dependency('gee-0.8'),
+    dependency('granite'),
+    dependency('gstreamer-1.0'),
+    dependency('gstreamer-pbutils-1.0'),
+    dependency('gtk+-3.0'),
+    dependency('glib-2.0'),
+    dependency('gobject-2.0'),
+    dependency('json-glib-1.0'),
+    dependency('libsecret-1'),
+    dependency('libsoup-2.4'),
+    dependency('libxml-2.0'),
+    dependency('sqlite3'),
+    dependency('webkit2gtk-4.0'),
+    # meson.get_compiler('vala').find_library('webkit2gtk-3.0', dirs: vapi_dir)
+]
+
+
+ss = ssmod.source_set()
+ss.add(src)
+ss.add(dependencies)
+ss.add(
+    when: dependency('unity'), 
+    if_true: dependency('unity'),
+)
+ss.add(
+    when: dependency('libnotify'), 
+    if_true: dependency('libnotify'),
+)
+config = configuration_data()
+ssconfig = ss.apply(config)
+
+i18n.merge_file(
+    input: 'data' / 'com.github.needleandthread.vocal.desktop.in',
+    output: meson.project_name() + '.desktop',
+    po_dir: meson.source_root() / 'po',
+    type: 'desktop',
+    install: true,
+    install_dir: get_option('datadir') / 'applications',
+)
+
+i18n.merge_file(
+    input: 'data' / 'com.github.needleandthread.vocal.appdata.xml.in',
+    output: meson.project_name() + '.appdata.xml',
+    po_dir: meson.source_root() / 'po',
+    install: true,
+    install_dir: get_option('datadir') / 'metainfo',
+)
+
+icon_sizes = ['16', '22', '24', '32', '48', '64', '128']
+
+foreach i : icon_sizes
+    install_data(
+        'icons' / i / 'com.github.needleandthread.vocal.svg',
+        install_dir: get_option('datadir') / 'icons' / 'hicolor' / i + 'x' + i / 'apps',
+        rename: meson.project_name() + '.svg',
```

Comment:
What is the purpose of renaming, everywhere?

Reply:
yeah, I might have followed the elementary developer docs too literally. but you are right rename is not needed.